trunk/SUAVE/Methods/Aerodynamics/AVL/create_avl_datastructure.py

```diff
@@ -23,214 +25,368 @@
+#              section.origin = origin[i_segs]
+#              avl_wing.append_section(section)
+#              
+#              # break condition for segment tip
+#              if (i_segs == n_segments-1):
+#                  return avl_wing
+#                  break        
+#              
+#                  # update origin for next segment
+#              segment_percent_span =    suave_wing.Segments[i_segs+1].percent_span_location - suave_wing.Segments[i_segs].percent_span_location     
+#       
+#              if avl_wing.vertical:
+#                  origin.append( [origin[i_segs][0] + semispan*segment_percent_span*np.tan(sweep[i_segs]),\
+#                                  origin[i_segs][1] + semispan*segment_percent_span*np.tan(dihedral[i_segs]),\
+#                                  origin[i_segs][2] + semispan*segment_percent_span])
+#              else:
+#                  origin.append( [origin[i_segs][0] + semispan*segment_percent_span*np.tan(sweep[i_segs]) ,\
+#                                  origin[i_segs][1] + semispan*segment_percent_span,\
+#                                  origin[i_segs][2] + semispan*segment_percent_span*np.tan(dihedral[i_segs])])
+#              
+#      return avl_wing
+#
+#-------------------------------------------------------------------------------------------------------------------------
+
 def populate_body_sections(avl_body,suave_body):
 
-	symm = avl_body.symmetric
-	semispan_h = avl_body.widths.maximum * 0.5 * (2 - symm)
-	semispan_v = avl_body.heights.maximum * 0.5
-	origin = suave_body.origin
-	root_section = Section()
-	root_section.tag    = 'center_horizontal_section'
-	root_section.origin = origin
-	root_section.chord  = avl_body.lengths.total
-	
-	tip_section = Section()
-	tip_section.tag  = 'outer_horizontal_section'
-	nl = avl_body.lengths.nose
-	tl = avl_body.lengths.tail
-	tip_section.origin = [origin[0]+nl,origin[1]+semispan_h,origin[2]]
-	tip_section.chord = root_section.chord - nl - tl
-	
-	avl_body.append_section(root_section,'horizontal')
-	avl_body.append_section(tip_section,'horizontal')
-	tip_sectionv1 = deepcopy(tip_section)
-	tip_sectionv1.origin[1] = origin[1]
-	tip_sectionv1.origin[2] = origin[2] - semispan_v
-	tip_sectionv1.tag       = 'lower_vertical_section'
-	avl_body.append_section(tip_sectionv1,'vertical')
-	avl_body.append_section(root_section,'vertical')
-	tip_sectionv2 = deepcopy(tip_section)
-	tip_sectionv2.origin[1] = origin[1]
-	tip_sectionv2.origin[2] = origin[2] + semispan_v
-	tip_sectionv2.tag       = 'upper_vertical_section'
-	avl_body.append_section(tip_sectionv2,'vertical')
-	
-	return avl_body
-	
+        symm = avl_body.symmetric   
+        semispan_h = avl_body.widths.maximum * 0.5 * (2 - symm)
+        semispan_v = avl_body.heights.maximum * 0.5
+        origin = [0, 0, 0]
+
+        # Compute the curvature of the nose/tail given fineness ratio. Curvature is derived from general quadratic equation
+        vec1 = [2 , 1.5, 1.2 , 1]
```

Comment:
I had to find a way to relate the fineness ratio to the quadratic curve formula. These vectors just contain points that I fitted then interpolated to get a curve.